### PR TITLE
should force load react-components which send over turbo-stream

### PR DESCRIPTION
**Pull request by theforestvn88**
### Summary

Support React On Rails w/Hotwire and Turbo Streams https://github.com/shakacode/react_on_rails/issues/1493, https://github.com/shakacode/react_on_rails/issues/1508

All turbo events (turbo:before-render, turbo:render,.. ) will not dispatch with TurboStream, except [turbo:before-stream-render](https://github.com/hotwired/turbo-rails/blob/main/app/assets/javascripts/turbo.js#L5071).
And it looks like turbo-rails did not support a after-stream-render event, i don't know why yet.

But hotwired send html, react_on_rails send script, right ?
So my solution: send a script that will trigger load react component along with it.

### Pull Request checklist

- i add a new render option: force_load which if true then the trigger load script will be concat with the component itself.

- i not use reactOnRailsPageLoaded which will query and reload all current react-components (not effective), i introduce new method reactOnRailsComponentLoaded(react-component-dom-id) to load directly the react-component that the script be sent along with.

### Other Information

- - -
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1620)

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `force_load` configuration option for React components, allowing immediate loading.
  - Added Turbo Frame and Turbo Stream support, enabling dynamic updates and enhanced interactivity.
  - Created a new React component `HelloTurboStream` that displays personalized greetings.
  - Integrated Turbo for improved navigation and page updates in the application.

- **Bug Fixes**
  - Ensured proper rendering behavior for Turbo Stream components.

- **Tests**
  - Added tests for new Turbo Stream functionality and `force_load` configuration.
  - Enhanced test coverage for immediate loading of the TurboStream component.

### CODE REVIEW

**Comment by justin808** [comment]
@theforestvn88 Please
1. Check the `coderabitai` suggestions
2. fix the CI failure on liniting
3. Rebase on master, and check CI failures.


@Judahmeek did you see anything requiring changes?

**Comment by justin808** [comment]
@theforestvn88 can you check if your PR solves the problem in https://github.com/shakacode/react_on_rails/issues/1508#issuecomment-1499730248?

File: lib/react_on_rails/configuration.rb [hunk 1]
```diff
@@ -68,7 +70,7 @@ def initialize(node_modules_location: nil, server_bundle_js_file: nil, prerender
                    same_bundle_for_client_and_server: nil,
                    i18n_dir: nil, i18n_yml_dir: nil, i18n_output_format: nil,
                    random_dom_id: nil, server_render_method: nil, rendering_props_extension: nil,
-                   components_subdirectory: nil, auto_load_bundle: nil)
+                   components_subdirectory: nil, auto_load_bundle: nil, force_load: nil)
```
[the remark this answers is not in the document]
Comment (theforestvn88) [reply lib/react_on_rails/configuration.rb:74]: Fixed, but add comment on line 43

File: spec/dummy/client/app/startup/HelloTurboStream.jsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import RailsContext from '../components/RailsContext';
```
[the remark this answers is not in the document]
Comment (theforestvn88) [reply spec/dummy/client/app/startup/HelloTurboStream.jsx:3]: Not Fix, it's be used later.

**Review by justin808 — APPROVED** [review]
@Judahmeek try this out.

@theforestvn88 looks great. sorry so long to get this reviewed!

**Comment by justin808** [comment]
Looks good.

**Comment by justin808** [comment]
@G-Rath any opinions on this PR?

**Comment by justin808** [comment]
@theforestvn88 can you fix the lint errors?

**Comment by theforestvn88** [comment]
> @theforestvn88 can you fix the lint errors?

i don't know what/where the lint errors are, the command `nps format.listDifferent` just return exit code 1.

Please help me !

**Review by justin808 — APPROVED** [review]

**Comment by justin808** [comment]
Need new task to update https://github.com/shakacode/react_on_rails/blob/master/docs/rails/turbolinks.md#using-turbo

Merging...

**Pull request merged**